AT3/AT3Tags.cpp

```diff
@@ -62,47 +109,221 @@ void AT3Tags::OnGetTagItem(CFlightPlan FlightPlan, CRadarTarget RadarTarget, int
 	strcpy_s(sItemString, 16, tagOutput.substr(0, 15).c_str());
 }
 
+void OnFunctionCall(int FunctionId, const char* sItemString, POINT Pt, RECT Area) {
+	CFlightPlan FlightPlan = AT3Tags().FlightPlanSelectASEL();
+	if (!FlightPlan.IsValid()) {
+		return;
+	}
+
+	string origin = FlightPlan.GetFlightPlanData().GetOrigin();
+	string SID = FlightPlan.GetFlightPlanData().GetSidName();
+	string dest = FlightPlan.GetFlightPlanData().GetDestination();
+	string destRunway = FlightPlan.GetFlightPlanData().GetArrivalRwy();
+	string STAR = FlightPlan.GetFlightPlanData().GetStarName();
+
+	string APPMenuName = dest + destRunway + "APPs";
+
+	switch (FunctionId) {
+		case TAG_FUNC_APP_SEL_MENU:
+			AT3Tags().OpenPopupList(Area, APPMenuName.c_str(), 1);
+		}
+}
+
 string AT3Tags::GetFormattedAltitude(CFlightPlan& FlightPlan, CRadarTarget& RadarTarget)
 {
-	return string();
+	int altitude = RadarTarget.GetPosition().GetPressureAltitude();
+	int transAlt = AT3Tags().GetTransitionAltitude();
```

Comment:
Same as above

Reply:
Good question, as you can tell I am not used to OOP in C++